.github/workflows/commit-codegen.yaml

```diff
@@ -0,0 +1,29 @@
+# Copyright 2023 SECO Mind Srl
+# SPDX-License-Identifier: Apache-2.0
+
+name: commit-codegen
+
+on:
+  workflow_call:
+  workflow_dispatch:
+
+permissions:
+  contents: write
+
+jobs:
+  commit:
+    name: Commit and push generated code
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Download rust proto
+        uses: actions/download-artifact@v3
+        id: download-rust
+        with:
+          name: rust-proto
+      - name: Move rust proto to src
+        run: |
+          mv -v '${{steps.download-rust.outputs.download-path}}/edgehog.device.forwarder.rs' \
+            rust/edgehog-device-forwarder-proto/src/proto.rs
+      - name: Commit the generated code
+        run: .github/workflows/scripts/commit.sh
```

Comment:
We are considering whether it is better to create a separate PR instead of doing direct push to the master branch... see here https://github.com/astarte-platform/astarte-message-hub-proto/pull/3/commits/8b44fcb6328d56a9ab681c5cab8a743bbdac6ff8#diff-47c440ec0726ad4c1976e840922373c6de7839fd55e82210c1b0e1eb93ce8d71

Reply:
I think it could be a good idea. I can implement the same thing